extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php

```diff
@@ -201,87 +195,118 @@ private function getUploadWarningMessages( $warnings ){
 		return $ret;
 	}
 
-	public function executeWordmarkUpload() {
-		// check rights
-		if ( !ThemeDesignerHelper::checkAccess() ) {
-			$this->displayRestrictionError( __METHOD__ );
-		}
+	public function wordmarkUpload() {
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+
+		// SUS-797: Validate edit token and POST request for external requests
+		$this->checkWriteRequest();
 
 		$upload = new UploadWordmarkFromFile();
 
 		$status = $this->uploadImage($upload);
 
 		if($status['status'] === 'uploadattempted' && $status['isGood']) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "uploadattempted" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "isGood" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)